src/v/kafka/security/tests/authorizer_test.cc

```diff
@@ -0,0 +1,846 @@
+      allow_write_acl,
+      deny_read_acl,
+    };
+
+    BOOST_REQUIRE(expected == get_acls(auth, prefixed_resource));
+    BOOST_REQUIRE(get_acls(auth, wildcard_resource).empty());
+    BOOST_REQUIRE(get_acls(auth, default_resource).empty());
+}
+
+BOOST_AUTO_TEST_CASE(auth_prefix_resource) {
+    acl_principal user(principal_type::user, "alice");
+    acl_host host("192.168.3.1");
+
+    authorizer auth;
+
+    auto add_acl = [&auth](ss::sstring name, pattern_type type) {
+        std::vector<acl_binding> bindings;
+        bindings.emplace_back(
+          resource_pattern(resource_type::topic, name, type), deny_read_acl);
+        auth.add_bindings(bindings);
+    };
+
+    auto add_lit = [&add_acl](ss::sstring name) {
+        add_acl(name, pattern_type::literal);
+    };
+
+    auto add_pre = [&add_acl](ss::sstring name) {
+        add_acl(name, pattern_type::prefixed);
+    };
+
+    add_lit("a_other");
+    add_pre("a_other");
+    add_pre("foo-" + random_generators::gen_alphanum_string(10));
+    add_pre("foo-" + random_generators::gen_alphanum_string(10));
+    add_pre("foo-" + random_generators::gen_alphanum_string(10) + "-zzz");
+    add_pre("fooo-" + random_generators::gen_alphanum_string(10));
+    add_pre("fo-" + random_generators::gen_alphanum_string(10));
+    add_pre("fop-" + random_generators::gen_alphanum_string(10));
+    add_pre("fon-" + random_generators::gen_alphanum_string(10));
+    add_pre("fon-");
+    add_pre("z_other");
+    add_lit("z_other");
+
+    BOOST_REQUIRE(!auth.authorized(
+      model::topic(default_resource.name()), acl_operation::read, user, host));
+
+    std::vector<acl_binding> bindings;
+    bindings.emplace_back(prefixed_resource, allow_read_acl);
+    auth.add_bindings(bindings);
+
+    BOOST_REQUIRE(auth.authorized(
+      model::topic(default_resource.name()), acl_operation::read, user, host));
+}
+
+BOOST_AUTO_TEST_CASE(single_char) {
+    acl_principal user(principal_type::user, "alice");
+    acl_host host("192.168.3.1");
+
+    authorizer auth;
+
+    std::vector<acl_binding> bindings;
+    bindings.emplace_back(
+      resource_pattern(resource_type::topic, "f", pattern_type::literal),
+      allow_read_acl);
+    auth.add_bindings(bindings);
+
+    BOOST_REQUIRE(
+      auth.authorized(model::topic("f"), acl_operation::read, user, host));
+
+    BOOST_REQUIRE(
+      !auth.authorized(model::topic("foo"), acl_operation::read, user, host));
+
+    bindings.clear();
+    bindings.emplace_back(
+      resource_pattern(resource_type::topic, "_", pattern_type::prefixed),
+      allow_read_acl);
+    auth.add_bindings(bindings);
+
+    BOOST_REQUIRE(
+      auth.authorized(model::topic("_foo"), acl_operation::read, user, host));
+
+    BOOST_REQUIRE(
+      auth.authorized(model::topic("_"), acl_operation::read, user, host));
+
+    BOOST_REQUIRE(
+      !auth.authorized(model::topic("foo_"), acl_operation::read, user, host));
+}
+
+BOOST_AUTO_TEST_CASE(get_acls_principal) {
```

Comment:
I wonder if we should port some of the acls of `aiokafka` to ducky for acls.

Reply:
https://github.com/vectorizedio/redpanda/issues/987